src/Bundle.ts

```diff
@@ -47,6 +47,11 @@ export default class Bundle {
 		const outputBundle = getOutputBundle(outputBundleBase);
 		this.pluginDriver.setOutputBundle(outputBundle, this.outputOptions);
 
+		const assetShouldBeReferenceKeys = Object.keys(outputBundle).filter(assetKey => {
+			const outputAsset = outputBundle[assetKey];
+			return outputAsset.type === 'asset' && outputAsset.needsCodeReference;
+		});
```

Comment:
This could probably moved next to the unreferenced asset logic below and both this block and lines 89-101 could be extracted into a helper function? E.g. `function removeUnreferencedAssets(outputBundle)`? That would make it more readable.

Reply:
Ok, will do.